### PR TITLE
plugin-catalog-backend-module-msgraph: add configuration to use search criteria to select groups

### PR DESCRIPTION
**Pull request by goenning**
Signed-off-by: Guilherme Oenning <goenning@eshopworld.com>

Adding optional `groupSearch` and `userGroupMemberSearch` configuration that maps to $search OData querystring which is used by Microsoft Graph API to perform extra filtering that is not supported by `groupFilter and `userGroupMemberFilter`

Just to add more context, we want to synchronise only groups that ends with `-team` in our org. It's not possible with filter, as only startsWith is supported. Although search is not perfect as it'll search anywhere on the string, it'll be enough to filter out unrelated groups.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend-module-msgraph/src/microsoftGraph/client.ts [hunk 1]
```diff
@@ -167,6 +172,11 @@ export class MicrosoftGraphClient {
     return await fetch(url, {
       headers: {
         Authorization: `Bearer ${token.accessToken}`,
+
+        // Eventual consistency is required to use $search.
+        // Groups/Users are not changed that frequently to require strong consistency
+        // If a new user/group is not found, it'll eventually be imported on a subsequent call
+        ConsistencyLevel: 'eventual',
```
Comment (freben) [inline plugins/catalog-backend-module-msgraph/src/microsoftGraph/client.ts:179]: would it be possible to set this _only_ if search is actually used?

**Review by freben — APPROVED** [review]
Looks good! Just one question I'd like to double check with you

File: plugins/catalog-backend-module-msgraph/src/microsoftGraph/client.ts (hunk 1 above)
Comment (goenning) [reply plugins/catalog-backend-module-msgraph/src/microsoftGraph/client.ts]: Yes. Done.

**Comment by freben** [comment]
Ach - seems like a `prettier` formatting error crept in :/

**Review by freben — APPROVED** [review]
Approving pre-emptively though, this looks good! Happy to merge after the formatting fix

**Comment by goenning** [comment]
Done! I am sure husky/precommit was working before, but I recently changed to a new machine and it doesn't seem to be doing its job 🤔

**Comment by freben** [comment]
ah but now the api report needs to be generated! :) sorry for the noise

**Comment by freben** [comment]
hey again, i rudely pushed the updated report on top - i hope you don't mind - let's see if it builds now!

**Comment by goenning** [comment]
Thanks for that push! I always forget that step! 🤦‍♂️

**Pull request merged**